photon-ml/src/main/scala/com/linkedin/photon/ml/FeatureIndexingJob.scala

```diff
@@ -19,79 +19,104 @@ import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.{FileSystem, Path}
 
 import scala.collection.mutable
+import scala.collection.JavaConverters._
 
-import com.linkedin.photon.ml.avro.{TrainingExampleFieldNames, ResponsePredictionFieldNames, FieldNames, AvroIOUtils}
+import com.linkedin.photon.ml.avro.{
+  AvroUtils, AvroIOUtils, FieldNames, ResponsePredictionFieldNames, TrainingExampleFieldNames}
 import com.linkedin.photon.ml.io.{FieldNamesType, GLMSuite}
-import com.linkedin.photon.ml.util.{PhotonLogger, Utils, PalDBIndexMapBuilder}
+import com.linkedin.photon.ml.util._
 import org.apache.avro.generic.GenericRecord
 import org.apache.spark.rdd.RDD
 import org.apache.spark.{HashPartitioner, SparkConf, SparkContext}
-
+import scopt.OptionParser
 
 import java.util.{List => JList}
 
 /**
   * A class that builds feature index map as an independent Spark job. Recommended when feature space is large,
   * typically when there are more than 200k unique features.
   *
-  * The job expected three required arguments and one optional one:
-  * [input_data_path]: The input path of data
-  * [partition_num]: The number of partitions to break the storage into. This is merely introduced as an optimization at
-  *   build stage that we don't necessary have to shuffle all features into the same partition but rather we could
-  *   just hold them into multiple ones. A heuristically good number could be 1-10, or even as large as 20; but please
+  * The job expects three required arguments and three optional ones:
+  *
+  * input-paths: The input paths of the data
+  * partition-num: The number of partitions to break the storage into. This is merely introduced as an optimization at
+  *   the building stage where we don't necessary have to shuffle all features into the same partition, but rather we
+  *   can hold them in multiple ones. A heuristically good number could be 1-10, or even as large as 20; but please
   *   avoid setting it into an arbitrarily large number. Use just 1 or 2 if you find the indexing job is already fast
   *   enough.
-  * [output_dir]: The output directory
-  * [if_add_intercept] (optional, default=true): whether include INTERCEPT into the map.
-  *
+  * output-path: The output path
+  * date-range: Date range for the input data represented in the form start.date-end.date, e.g. 20150501-20150631
+  * date-range-days-ago: Date range for the input data represented in the form start.daysAgo-end.daysAgo, e.g. 90-1
+  * add-intercept: whether include INTERCEPT in the map (optional, default: true).
+  * data-format: Input data format. (optional, default: TRAINING_EXAMPLE, options:
```

Comment:
It looks like this is documenting the command line api rather than the class. I don't feel strongly about either keeping it that way or changing it.

@ashelkovnykov never!